src/hooks/useEffectAsync.ts

```diff
@@ -0,0 +1,37 @@
+import React, { useEffect } from 'react';
+
+type AsyncEffect = (signal: AbortSignal) => Promise<void | (() => void)> | void;
+
+/**
```

Comment:
Could you add proper JS doc format param description as well here?
```ts
/**
 * 
 * @param effect 
 * @param deps 
 */
```

Reply:
Used the official React jsdoc for useeffect and added the description of the optional abortsignal usage. Good enough?

Reply:
Perfect, thanks! 👍